MAINTENANCE.md

```diff
@@ -0,0 +1,294 @@
+
+### Common steps
+
+1. Verify that documentation book was published at [Documentation book](https://rust-driver.docs.scylladb.com/stable/).
+2. Publish all updated crates to crates.io. Make sure you are using newest stable version of Rust. In the main folder of the repository run `cargo publish -p <crate>` for each crate that is being updated, in correct order (`scylla-macros`, `scylla-cql`, `scylla-proxy`, `scylla`).
+3. Verify that new versions are visible on crates.io and their docs on docs.rs.
+4. Go to https://github.com/scylladb/scylla-rust-driver/releases , click the `Draft new release` button and follow the procedure to create a new release on GitHub. Use the release notes as its description.
+5. (Mandatory for major release, optional for minor release) Publish a post on the forum:
+    - Go to [Release notes](https://forum.scylladb.com/c/scylladb-release-notes/18) section.
+    - Click "New Topic".
+    - Title should be `[RELEASE] ScyllaDB Rust Driver <version>`, e.g. `[RELEASE] ScyllaDB Rust Driver 0.14.0`
+    - Tags: `release`, `drivers`, `rust-driver`, `rust`.
+    - Content of the post should just be release notes.
+    - Click "Create Topic"
+    - Posts in "Release notes" section need additional confirmation. You can write to current forum admin to expedite this.
+
+You're done!
+
+
+## Writing release notes
+
+PR titles are written for maintainers, but release notes entries are written for users.
+It means that they should not be the same - and if they are then they are probably
+not good as either PR titles or release notes entries.
+
+For that reason we hand-write our release notes, and link to relevant PRs in the entries.
+
+Some old release notes that you can take inspiration from when writing new ones:
+- https://github.com/scylladb/scylla-rust-driver/releases/tag/v0.10.0
+- https://github.com/scylladb/scylla-rust-driver/releases/tag/v0.14.0
+- https://github.com/scylladb/scylla-rust-driver/releases/tag/v0.11.0
+- https://github.com/scylladb/scylla-rust-driver/releases/tag/v0.13.0
+
+The template for release notes can be found in the section below.
+
+Guidelines on how to write release notes:
+
+- Go over all the PRs / commits since previous release. Usually: `git log --first-parent` to see
+  merge commits and commits that are directly on a branch. You can also try filtering
+  merged PRs on Github by merge date, but it's cumbersome.
+
+- Items in release notes will usually correspond 1:1 to PRs / comits - but not always. It is possible that
+  some functionality that should be a single item on the list is split over multiple PRs.
+  It is also possible that single PR will be mentioned in two items.
+
+- Release notes for a version should only contain items that make it different
+  from previous version. Example: commit `A` that is on main was backported to
+  `branch-0.13.x`, and `v0.13.2` was released with it - then release `0.14` will
+  not include `A` in release notes, but `0.13.2` will.
```

Comment:
Let's consider the following scenario. A user decides to update the driver to a next major version (e.g. 0.14). Thus, they read the release notes for that release. However, as minor dependency updates happens automatically with Cargo, they miss the release notes for the previous minor release (0.13.2) that brought some crucial changes/bugfixes.

To prevent such scenario, let's include all changes that happened since last major release in next major release's notes.

Reply:
It will cause duplication of entries, but will make it easier to write release notes. I think this tradeoff is acceptable.

Reply:
Of course we will need to change a lot of those procedures / rules when 1.0 is released.